frontend/site/.vitepress/theme/components/InventoryBase.vue

```diff
@@ -0,0 +1,276 @@
+<template>
+  <div>
+    <div v-if="datasets.length > 0">
+      <InventoryFilter
+        @change="updateFilters"
+        :organizations="organizations"
+        :statuses="statuses"
+        :trimesters="trimesters"
+        :types="types"
+        ref="filtersComponent"
+      />
+      <div class="fr-table fr-table--no-caption">
+        <InventoryTable :datasets="filteredSortedDatasets" :columns="mapping" />
+        <p v-if="nextCursor" class="fr-mt-2w">
+          <button @click="loadMore" class="fr-btn fr-btn--sm">Charger plus de résultats</button>
+        </p>
+      </div>
+
+      <p v-if="lastModified">Dernière modification : {{ formatDate(lastModified) }}</p>
+    </div>
+    <div v-else>
+      <em>Données en cours de chargement...</em>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import dayjs from "dayjs";
+import quarterOfYear from "dayjs/plugin/quarterOfYear";
+import { computed, reactive, ref } from "vue";
+import useDataProxy from "../composables/dataProxy";
+import InventoryFilter from "./InventoryFilter.vue";
+import InventoryTable from "./InventoryTable.vue";
+
+dayjs.extend(quarterOfYear);
+
+/**
+ * @type {Array<import("../types").Column>}
+ */
+const mapping = [
+  {
+    key: "type",
+    label: "Type",
+    source: "Données, API ou code source",
+  },
+  {
+    key: "title",
+    label: "Titre",
+    source: "Jeux de données",
+    format: (cell, row) =>
+      row.raw.URL ? `<a href="${row.raw.URL}">${cell}</a>` : cell,
+    transform: (cell) => cell[0].plain_text,
+  },
+  {
+    key: "organization",
+    label: "Ministère",
+    source: "Nom Producteur",
+    transform: (cell) => cell.string,
+  },
+  {
+    key: "status",
+    label: "Statut d’ouverture",
+    source: "Statut",
+    format: (cell, row) =>
+      `<span class="fr-tag ${row.status._class}">${cell.label}</a>`,
+    width: "12em",
+  },
+  {
+    key: "date",
+    label: "Date estimée de publication",
+    source: "Calendrier ouverture",
+    format: (cell, row) => {
+      const date = row.date;
+      if(!date) {
+        return "";
+      }
+      const [year, trimester] = date.split(" ");
+      const now = dayjs();
+
+      if (
+        row.status.key != "open" &&
+        now.year() >= parseInt(year) &&
+        now.quarter() > parseInt(trimester[1])
+      ) {
+        return `T${now.quarter()} ${now.year()}`;
+      } else {
+        return `${trimester} ${year}`;
+      }
+    },
+    transform: (cell) => cell?.name,
+  },
+];
+
+
+
+/**
+ * @type {Array<import("../types").Status>}
+ */
+const statuses = [
```

Comment:
I find the source lists here a bit dangerous to maintain, what if a new label is created in the Notion base?

Reply:
There is a `public` checkbox defaulting to false in the base, so only chosen rows are shown here.

We can add a default value to avoid an error if the Notion list of statuses is updated before the project and a row with a new status is marked as `public`.

We can remove these sources here by doing the mapping in a Notion formula instead, it's not easier to find but it can be done without publishing a project update.

Reply:
We'll try to remove this statuses mapping and use a Notion formula instead and remove most of the mapping array too.